### PR TITLE
@types/polka - Support instantiated server that exists on Polka API

### LINKED ISSUE #50440: Missing type support for `server` in @types/polka
- [x] I tried using the `@types/polka` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @pkuczynski @JamesMessinger

### Missing support for ``server`` in index.d.ts

``server`` exists, possibly as undefined or as http.Server, on an an instance of `polka()` as per the [Polka docs](https://github.com/lukeed/polka#listen) but is not exposed by the types for polka.

[PR to resolve type issue](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50441)

### PR DESCRIPTION
**Pull request by barclayd**
Fixes https://github.com/DefinitelyTyped/DefinitelyTyped/issues/50440

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Passing tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/lukeed/polka#optionsserver

### CODE REVIEW

File: types/polka/index.d.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@
 // Definitions by: Piotr Kuczynski <https://github.com/pkuczynski>
 //                 James Messinger <https://github.com/JamesMessinger>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// Minimum TypeScript Version: 3.1
+// Minimum TypeScript Version: 3.7
```
Comment (pkuczynski) [inline types/polka/index.d.ts:6]: Why would you bump minimym TS version?

**Review by pkuczynski — CHANGES_REQUESTED** [review]

File: types/polka/index.d.ts (hunk 1 above)
Comment (barclayd) [reply types/polka/index.d.ts:6]: Hi @pkuczynski, thanks for the great work you've done with these types.
I got errors when running the tests locally without bumping the version to 3.7 
Screenshot 2021-01-06 at 21 13 29


Same error also appeared in [CI test run](https://github.com/DefinitelyTyped/DefinitelyTyped/runs/1658058249#step:5:97)

Comment (pkuczynski) [reply types/polka/index.d.ts:6]: Makes sense! 👍

File: types/polka/index.d.ts [hunk 2]
```diff
@@ -95,6 +95,13 @@ declare namespace polka {
          * It receives all requests and tries to match the incoming URL against known routes.
          */
         handler(req: Request, res: ServerResponse, parsed?: Url): void;
+
+        /**
+         * The instantiated `server` Polka creates when `listen()` is called.
+         * `server` is only created if a server was not provided via `option.server`
+         * `server` will be undefined until polka.listen is invoked or if a server was provided.
+         */
+        server?: Server;
```
Comment (pkuczynski) [inline types/polka/index.d.ts:104]: In the docs https://github.com/lukeed/polka I do not see `server` to be exposed?

Comment (barclayd) [reply types/polka/index.d.ts:104]: `server` is exposed in the code snippet in the code example provided [here](https://github.com/lukeed/polka#listen) in the polka README.

From Polka README:

```javascript
// README.md
const { server, handler } = polka().listen();

// Or this!
const app = polka().listen(PORT, onAppStart);

app.use('users', require('./users'))
  .get('/', (req, res) => {
    res.end('Pretty cool!');
  });
```

`server` also exists on the Polka class as shown [here](https://github.com/lukeed/polka/blob/master/packages/polka/index.js#L33)

Comment (pkuczynski) [reply types/polka/index.d.ts:104]: Makes sense! Thanks! :)

**Review by pkuczynski — APPROVED** [review]

File: types/polka/index.d.ts (hunk 2 above)
Comment (barclayd) [reply types/polka/index.d.ts:104]: No problem, thanks for the approval 😊

**Comment by barclayd** [comment]
Ready to merge

**Comment by barclayd** [comment]
Thank you @typescript-bot

**Pull request merged**